### PR TITLE
ss-15: Add Dropdown

### PR DESCRIPTION
**Pull request by MstyslavDmytryk**
single select dropdown component

dropdown

### CODE REVIEW

**Review by ValeriiArt — APPROVED** [review]
На перший погляд все ок

File: frontend/src/bundles/common/components/dropdown/dropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import React, { useCallback } from 'react';
+import Select from 'react-select';
+
+import { type dataTypes } from '../../types/dropdown.type';
+import styles from './styles.module.scss';
+
+interface Properties {
+    data: dataTypes[];
```
Comment (VItaliy-Kalenichenko) [inline frontend/src/bundles/common/components/dropdown/dropdown.tsx:8]: Types should be uppercased

File: frontend/src/bundles/common/components/dropdown/dropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    selectedOption: dataTypes;
+    handleChange?: (option: dataTypes | null) => void;
+    width?: string | number;
```
Comment (VItaliy-Kalenichenko) [inline frontend/src/bundles/common/components/dropdown/dropdown.tsx:11]: Use one type for such cases

File: frontend/src/bundles/common/components/dropdown/dropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                name: selectedOption.name,
+                image: selectedOption.image,
+            }}
+            onChange={handleChange}
+            options={options}
+            formatOptionLabel={formatOptionLabel}
+            styles={{
+                container: (provided) => ({
+                    ...provided,
+                    width: typeof width === 'number' ? `${width}px` : width,
```
Comment (VItaliy-Kalenichenko) [inline frontend/src/bundles/common/components/dropdown/dropdown.tsx:56]: Update the prop type and this condition will lose its relevance

**Comment by MstyslavDmytryk** [comment]
photo

File: frontend/src/bundles/common/components/dropdown/dropdown.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    return (
+        <Select
+            className={styles.select}
+            value={{
+                value: selectedOption.value,
+                name: selectedOption.name,
+                image: selectedOption.image,
+            }}
+            onChange={
+                handleChange as unknown as (
```
Comment (VItaliy-Kalenichenko) [inline frontend/src/bundles/common/components/dropdown/dropdown.tsx:104]: Create an interface for this function. Try to avoid using 'unknown' type

**Review by VItaliy-Kalenichenko — APPROVED** [review]

**Pull request merged**